drivers/gles2/rasterizer_scene_gles2.cpp

```diff
@@ -880,7 +880,10 @@ RID RasterizerSceneGLES2::light_instance_create(RID p_light) {
 
 	light_instance->light_index = 0xFFFF;
 
-	ERR_FAIL_COND_V(!light_instance->light_ptr, RID());
+	if (!light_instance->light_ptr) {
+		memdelete(light_instance);
+		ERR_FAIL_V(RID());
```

Comment:
In this case the condition is meaningful, as it will appear in the error message. After this change, we'd just have a plain error message with no mention of `light_instance->light_ptr` being NULL.

Reply:
I think that ERR_EXPLAIN("!light_instance->light_ptr") should be enough.

Reply:
I don't know, it should either be a proper English sentence in `ERR_EXPLAIN()`, or the auto-generated `Condition ' !light_instance->light_ptr ' is true.` which is familiar to developers.

Reply:
I still think it's pointless to reimplement the error message that was already working just fine simply to satisfy a false positive static check, but fair enough, that shouldn't block the rest of those changes.

Reply:
But the static analyzer didn't report a false error, but a memory leak and usage of memory after free.

Code in drivers/gles2/rasterizer_scene_gles2.cpp caused memory leak because memory wasn't freed after error.
```
LightInstance *light_instance = memnew(LightInstance);
...
ERR_FAIL_COND_V(!light_instance->light_ptr, RID()); //LEAK
...
```
Code in drivers/gles3/rasterizer_scene_gles3.cpp caused reading freed data
```
if (!light_instance->light_ptr) {
	memdelete(light_instance); // Remove light_instance
	ERR_FAIL_COND_V(!light_instance->light_ptr, RID()); //Read from removed memory
```

Reply:
Ah I see, thanks for the clarification. Seems good then.